src/discord.ts

```diff
@@ -0,0 +1,140 @@
+import { Client, GuildMember, TextChannel } from "discord.js";
+import { subscribeToSelector, dispatch } from "./store";
+import { createSelector } from "@reduxjs/toolkit";
+import {
+  getOnlineUsers,
+  getHasSeenStart,
+  getIsActive,
+  getLastZoomTime,
+  setLastZoomTime,
+} from "./store/zoom";
+import { DISCORD_ACTIVE_ROLE, ZOOM_TIME_THRESHOLD, ZOOM_TIME_ANNOUNCEMENT_CHANNEL, ZOOM_TIME_DEBOUNCE_HOURS } from "./config";
+import { formatMessage } from "./helpers";
+
+const getActiveDiscordUsers = createSelector(
+  getOnlineUsers,
+  (onlineUsers) => {
+    return onlineUsers.map((user) => user.discordId).filter(Boolean);
+  }
+);
+
+const getZoomInfo = createSelector(
```

Comment:
nit: can we rename this to `getZoomStatusInfo` for consistency with `getZoomTimeInfo`?